pkg/spao/mac.go

```diff
@@ -103,17 +139,39 @@ func initCMAC(key []byte) (hash.Hash, error) {
 	return mac, nil
 }
 
+// @ requires  len(buf) >= MACBufferSize
+// @ preserves sl.Bytes(buf, 0, len(buf))
+// @ preserves sl.Bytes(ubuf, 0, len(ubuf))
+// @ preserves s != nil
+// @ preserves acc(s, R9)
+// @ preserves acc(s.Mem(ubuf), R8)
+// @ preserves acc(s.Path.Mem(ubuf), R49)
+// @ preserves s.ValidPathMetaData(ubuf)
+// @ preserves acc(opt.EndToEndOption, R48)
+// @ preserves len(opt.OptData) >= 12
+// @ preserves acc(sl.Bytes(opt.OptData, 0, len(opt.OptData)), R49)
+// @ preserves acc(sl.Bytes(pld, 0, len(pld)), R50)
+// @ ensures   0 <= n && n <= MACBufferSize
+// @ ensures   retErr != nil ==> retErr.ErrorMem()
+// @ decreases
 func serializeAuthenticatedData(
```

Comment:
I will review `serializeAuthenticatedData` and `zeroOutMutablePath` and the other non-reviewed files after my first comments are answered